python/Ganga/Core/GangaRepository/SessionLock.py

```diff
@@ -533,7 +533,7 @@ def session_read(self, fn):
             # This can fail (thats OK, file deleted in the meantime)
             fd = None
             try:
-                fd = os.open(fn, os.O_RDONLY)
+                fd = os.open(fn, os.O_RDWR)
```

Comment:
@rob-c This is an attempt to fix the problems mentioned in #959. I have indeed changed more places to open the file in r/w mode instead. Is there any performance issue related to opening a file in r/w mode relative to just read mode or just write mode?